nservicebus/hosting/critical-errors.md

```diff
@@ -46,12 +46,22 @@ snippet:CustomHostErrorHandlingAction
 
 ## When to override the default action
 
-The default action should be overridden whenever that default does not meet the specific hosting requirements. For example
+When self hosting the critical error callback must be overriden in order to exit the current process. Not overriding the critical error callback will result in the endpoint instance to stop processing messages without the ability to recover from this state.
 
- * If using [NServiceBus Host](/nservicebus/hosting/nservicebus-host), and wish to take a custom action before the endpoint process is killed.
- * If self hosting the process can be shut down the process via [Environment.FailFast](https://msdn.microsoft.com/en-us/library/dd289240.aspx) and re-start the process once the root cause has been diagnosed.
+NOTE: If not killing the process and just disposing the bus, be aware that any `Send` operations will result in [ObjectDisposedException](https://msdn.microsoft.com/en-us/library/system.objectdisposedexception.aspx) being thrown.
+
+When the [NServiceBus Host](/nservicebus/hosting/nservicebus-host) is used it is not required to override this callback as the host has a default implementation that exits the process. However, if it is required to execute tasks just before shutdown then it can be overriden in the same way as when self hosting.
+
+Consider the following things when implementing a critical error callback:
+
+- Wrap any tasks in a `try..finally`, making sure the process will exit if any tasks fail.
```

Comment:
note that this is redundant of people follow the `Environment.FailFast` direction below